src/Auth.ts

```diff
@@ -704,22 +704,111 @@ export class Auth {
       await logger.logToStderr('Trying to retrieve access token using federated identity...');
     }
 
-    if (!process.env.ACTIONS_ID_TOKEN_REQUEST_URL || !process.env.ACTIONS_ID_TOKEN_REQUEST_TOKEN) {
-      throw new CommandError('Federated identity is currently only supported in GitHub Actions.');
+    if (process.env.ACTIONS_ID_TOKEN_REQUEST_URL && process.env.ACTIONS_ID_TOKEN_REQUEST_TOKEN) {
+      if (debug) {
+        await logger.logToStderr('ACTIONS_ID_TOKEN_REQUEST_URL and ACTIONS_ID_TOKEN_REQUEST_TOKEN env variables found. The context is GitHub Actions...');
+      }
+
+      const federationToken = await this.getFederationTokenFromGithub(logger, debug);
+      return this.getAccessTokenWithFederatedToken(resource, federationToken, logger, debug);
+    }
+    else if (process.env.SYSTEM_OIDCREQUESTURI) {
+      if (debug) {
+        await logger.logToStderr('SYSTEM_OIDCREQUESTURI env variable found. The context is Azure DevOps...');
+      }
+
+      if (!process.env.SYSTEM_ACCESSTOKEN) {
+        throw new CommandError(`The SYSTEM_ACCESSTOKEN environment variable is not available. Please check the Azure DevOps pipeline task configuration. It should contain 'SYSTEM_ACCESSTOKEN: $(System.AccessToken)' in the env section.`);
+      }
+
+      const serviceConnectionId = process.env.AZURESUBSCRIPTION_SERVICE_CONNECTION_ID;
+      const serviceConnectionAppId = process.env.AZURESUBSCRIPTION_CLIENT_ID;
+      const serviceConnectionTenantId = process.env.AZURESUBSCRIPTION_TENANT_ID;
+      const useServiceConnection = serviceConnectionId && serviceConnectionAppId && serviceConnectionTenantId;
+
+      if (debug) {
+        if (!useServiceConnection) {
+          await logger.logToStderr('Not using a service connection. Run this command in an AzurePowerShell task to be able to use a service connection.');
+
+          if (!this.connection.appId || this.connection.tenant === 'common') {
+            throw new CommandError('The appId and tenant parameters are required when not using a service connection.');
+          }
+        }
+        else {
+          if (this.connection.appId || this.connection.tenant !== 'common') {
+            await logger.logToStderr('When using a service connection, the appId and tenant values are updated to the values of the service connection.');
+          }
+
+          this.connection.appId = serviceConnectionAppId;
+          this.connection.tenant = serviceConnectionTenantId;
+
+          await logger.logToStderr(`Using service connection '${serviceConnectionId}' with app Id '${serviceConnectionAppId}' and tenant Id '${serviceConnectionTenantId}'...`);
+        }
+      }
+
```

Comment:
the 
```
this.connection.appId = serviceConnectionAppId;
this.connection.tenant = serviceConnectionTenantId;
```

Should be outside of debug. 
As over our chat from the screenshots I passed on. With Debug it works perfectly.. without it gets 400